aws-lc-rs/src/ec.rs

```diff
@@ -265,6 +266,23 @@ unsafe fn validate_evp_key(
     Ok(())
 }
 
+#[allow(clippy::cast_ptr_alignment)]
+pub(crate) fn marshal_x509_public_key_to_buffer(
+    buffer: &mut [u8; PUBLIC_KEY_MAX_LEN],
+    x509_pubkey: &LcPtr<X509_PUBKEY>,
+) -> Result<usize, Unspecified> {
+    let evp_pkey = LcPtr::new(unsafe { X509_PUBKEY_get(**x509_pubkey) })?;
+    let bio = LcPtr::new(unsafe { BIO_new(BIO_s_mem()) })?;
+    if unsafe { i2d_PUBKEY_bio(*bio, *evp_pkey) } <= 0 {
+        return Err(Unspecified);
+    }
+    let size = BIO_get_mem_data(*bio, buffer.as_mut_ptr().cast::<*mut c_char>());
```

Comment:
This cast() looks suspicious to me -- again caveats of not looking at the sources, but generally it feels weird to cast a `*mut u8` to `*mut *mut c_char`, it seems like this is likely going to end up writing to some arbitrary memory address:

> BIO_get_mem_data() sets *pp to a pointer to the start of the memory BIOs data

(https://www.openssl.org/docs/man3.1/man3/BIO_get_mem_data.html)

BIO_get_mem_data should really be an `unsafe` function too, not sure why it isn't...

Reply:
Will fix this and rebase to the latest main.